github4s/src/main/scala/github4s/Github.scala

```diff
@@ -16,19 +16,18 @@
 
 package github4s
 
-import java.util.concurrent.TimeUnit.MILLISECONDS
-import cats.effect.ConcurrentEffect
+import cats.effect.Sync
 import github4s.algebras._
 import github4s.modules._
-import scala.concurrent.ExecutionContext
-import scala.concurrent.duration.Duration
+import org.http4s.client.Client
 
-class Github[F[_]: ConcurrentEffect](accessToken: Option[String], timeout: Option[Duration])(
-    implicit ec: ExecutionContext
+class Github[F[_]: Sync](
+    client: Client[F],
+    accessToken: Option[String]
 ) {
 
   private lazy val module: GithubAPIs[F] =
-    new GithubAPIv3[F](accessToken, timeout.getOrElse(Duration(1000L, MILLISECONDS)))
+    new GithubAPIv3[F](client, accessToken)
 
   lazy val users: Users[F]                 = module.users
```

Comment:
Not related to this PR, but is there any reason for all this stuff to be lazy?

Reply:
I guess if you're using only one module, the others don't get evaluated? :thinking: